src/Linters/HHClientLintError.hack

```diff
@@ -0,0 +1,73 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST;
+
+enum ArcanistLintSeverityEnum: string as string {
+  ADVICE = 'advice';
+  AUTOFIX = 'autofix';
+  WARNING = 'warning';
+  ERROR = 'error';
+  DISABLED = 'disabled';
+}
+
+final class HHClientLintError implements LintError {
+
+  const type TJSONError = shape(
+    'descr' => string,
+    'severity' => ArcanistLintSeverityEnum,
+    'path' => string,
+    'line' => int,
+    'start' => int,
+    'end' => int,
+    'code' => int,
+    'bypass_changed_lines' => bool,
+    'original' => string,
+    'replacement' => string,
+  );
+
+  public function __construct(
+    private File $file,
+    private this::TJSONError $error,
+    private ?string $blame_code = null
+  ) {
+  }
+
+  public function getFile(): File {
+    return $this->file;
+  }
+
+  public function getDescription(): string {
+    return $this->error['descr'];
+  }
+
+  public function getPosition(): (int, int) {
+    return tuple($this->error['line'], $this->error['end']);
+  }
+
+  public function getRange(): ((int, int), (int, int)) {
+    return tuple(
+      tuple($this->error['line'], $this->error['start']),
+      tuple($this->error['line'], $this->error['end']),
+    );
+  }
+
+  public function getBlameCode(): ?string {
+    return $this->blame_code;
+  }
+
+  public function getPrettyBlame(): ?string {
+    return $this->getBlameCode();
+  }
+
+  public function getLintRule(): LintRule {
```

Comment:
this should probably be memoized or passed in to the constructor given that it's never going to change for a particular instance

Reply:
By the way, Hotspot JVM is good at optimizing allocation for short-lived objects like this. Is HHVM able to eliminate the allocation like JVM does?

Reply:
Sorry, I'm not familiar enough with either VM:)